src/ECharts/Series.purs

```diff
@@ -0,0 +1,349 @@
+module ECharts.Series where
+
+import Control.Monad.Eff
+import Data.Function
+import Data.Maybe
+import Data.Argonaut.Core
+import Data.Argonaut.Encode
+import Data.Argonaut.Combinators
+import Data.Tuple
+import Data.StrMap
+
+import ECharts.Common
+import ECharts.Coords
+import ECharts.Chart
+import ECharts.Tooltip
+import ECharts.Type
+import ECharts.Style.Item
+import ECharts.Mark.Line
+import ECharts.Mark.Point
+import ECharts.Item.Data
+import ECharts.Symbol
+import ECharts.Series.Force
+import ECharts.Series.Gauge
+import ECharts.Series.EventRiver
+import ECharts.Axis
+import ECharts.Title
+
+newtype Series =
+  Series {
```

Comment:
This is a product encoding of a sum type, which makes sense in Javascript but not in Purescript, which has first-class sum types. Instead, this should be written as:

```purescript
data Series = Universal UniversalRec | Cartesian CartesianRec | ...
```

Where `UniveralRec`, `CartesianRec`, etc., are type synonyms for record types which contain the appropriate fields. Then the `EncodeJson` for this sum type can inject the appropriate `type` field into the serialized Json object.